FieldKit/app/services/db-interface.js

```diff
@@ -305,29 +308,71 @@ export default class DatabaseInterface {
         });
     }
 
-    _updateStream(station, status, type, index) {
-        if (!station.deviceId && station.device_id) {
-            station.deviceId = station.device_id;
+    getStationByStationIds(ids) {
+        return this.getDatabase().then(db => db.query(`SELECT * FROM streams WHERE station_id IN ($1)`, ids)).then(rows => {
+            return sqliteToJs(rows);
+        });
+    }
+
+    getStreamsByStationId(stationId) {
+        return this.getDatabase().then(db => db.query(`SELECT * FROM streams WHERE station_id = $1`, stationId)).then(rows => {
+            return sqliteToJs(rows);
+        });
+    }
+
+    getStreamsByStation(station) {
```

Comment:
Just curious about keeping both of these methods? (getStreamsByStationId and getStreamsByStation) Why not choose one and use only it?

Reply:
I leave them around just in case people need one in the future.